pkg/cmd/taskrun/delete.go

```diff
@@ -147,6 +147,7 @@ func allTaskRunNames(cs *cli.Clients, keep int, ns string) ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
+	reverse(taskRuns)
```

Comment:
The list of TaskRuns is not sorted by start time, so reversing the list will not work. You need to actually sort the TaskRuns using the [`sortByStartTime`](https://github.com/tektoncd/cli/blob/f2b65dd2be25975ec91c702811f1d101dc052d64/pkg/cmd/taskrun/list.go#L207) func. `reverse` should not be needed at all as you can just access the last element of the slice.

Reply:
I had tested this incorrectly which made me opt for the reverse. I have reverted back to the initial sorting I did in https://github.com/tektoncd/cli/commit/2778b104684e3e2161cdd642b10020b431feb30a